src/os/Transaction.h

```diff
@@ -299,15 +307,15 @@ class Transaction {
   /* Operations on callback contexts */
   void register_on_applied(Context *c) {
```

Comment:
maybe clarify the ownership xfer?

Reply:
I was looking for that and there are numerous callers. Still, refactoring them seems doable (follow up PR?).

Reply:
Sorry - I just meant adding a clear comment for the callers.

Reply:
Ah, sure!